### ISSUE TITLE
The $domain parameter must be a single text  string literal. Found: self::$text_domain

### ISSUE THREAD

**Issue by abhij89** [WordPress-Coding-Standards/WordPress-Coding-Standards#2398]
## Bug Description

I am using phpcs with standard WordPress. I have defined the domain globally in my class as a variable. Then using the same variable to pass the domain to all escaping functions. But it is throwing error saying `The $domain parameter must be a single text string literal. Found: self::$text_domain` which is not correct

## Minimal Code Snippet

`// Defined variable`
`public static $text_domain = 'my-plugins-domain';`

The issue happens when running this command:
```bash
phpcs --standard=WordPress my-plugin-file.php
```

... over a file containing this code:
```php
<?php _e( 'My Plugin Title', self::$text_domain ); ?>
```

The file was not auto-fixed via `phpcbf`

## Environment

| Question                 | Answer
| ------------------------ | -------
| PHP version              | 8.1.0
| PHP_CodeSniffer version  | 3.7.2
| WordPressCS version      | 3.0.1
| WordPressCS install type | Composer global

## Tested Against `develop` Branch?
- [ x ] I have verified the issue still exists in the `develop` branch of WordPressCS.

**Comment by jrfnl** [comment]
@abhij89 I'm not sure what you are reporting, but the sniff is working exactly as it should.

You should NEVER put the text domain in a variable or constant and pass that. You should ALWAYS pass a literal text string as the text domain.


There have been numerous articles written about this. I suggest you read them.

Some suggestions:
* https://markjaquith.wordpress.com/2011/10/06/translating-wordpress-plugins-and-themes-dont-get-clever/
* https://pascalbirchler.com/text-domain-wordpress-internationalization/

**Comment by abhij89** [comment]
My bad, I missed this rule.

**Comment by abhij89** [comment]
Thank you for clarifying
